ONSdigital/dp-search-reindex-batch
# Use feature flags to enable or disable the dataset api and zebedee

### What

When running the Search Reindex Batch for development and testing purposes it would be convenient to remove its dependencies on Zebedee and/or the Dataset API. These changes allow those dependencies to be switched on or off, which effectively means that the script can reindex documents that come from one or other, or neither, of those services, or both of those services (the default).

### How to review

- ```export SERVICE_AUTH_TOKEN=<local token value>```
- run zebedee locally: ```./run.sh```
- run MongoDB locally e.g. using dp-compose
- make sure that you have some zebedee collections (in zebedee-content) and some dataset api instances, editions, and datasets (in Mongo DB).
-  ```export DISABLE_GRAPH_DB_DEPENDENCY=true```
- run the dataset api locally: ```make debug```
- clean the search stack and start it up:
```
cd dp-compose/v2/stacks/search
make clean
make up
```
- run the search reindex batch with its default settings (both zebedee and dataset api enabled):
```
cd dp-search-reindex-batch
make debug
```

Then try it with different settings e.g.

Enable dataset api only:
export ENABLE_ZEBEDEE_REINDEX=false
export ENABLE_DATASET_API_REINDEX=true

Enable zebedee only:
export ENABLE_ZEBEDEE_REINDEX=true
export ENABLE_DATASET_API_REINDEX=false

Both disabled:
export ENABLE_ZEBEDEE_REINDEX=false
export ENABLE_DATASET_API_REINDEX=false

### Who can review

Anyone but me.
